src/OpenTelemetry.Resources.Process/ProcessResourceBuilderExtensions.cs

```diff
@@ -17,8 +17,32 @@ public static class ProcessResourceBuilderExtensions
     /// <param name="builder">The <see cref="ResourceBuilder"/> being configured.</param>
     /// <returns>The instance of <see cref="ResourceBuilder"/> being configured.</returns>
     public static ResourceBuilder AddProcessDetector(this ResourceBuilder builder)
+    {
+        return builder.AddProcessDetector(new ProcessDetectorOptions());
+    }
+
+    /// <summary>
+    /// Enables process resource detector.
+    /// </summary>
+    /// <param name="builder">The <see cref="ResourceBuilder"/> being configured.</param>
+    /// <param name="options">The <see cref="ProcessDetectorOptions"/> which controls the behavior of the resource detector.</param>
+    /// <returns>The instance of <see cref="ResourceBuilder"/> being configured.</returns>
+    public static ResourceBuilder AddProcessDetector(this ResourceBuilder builder, ProcessDetectorOptions options)
     {
         Guard.ThrowIfNull(builder);
-        return builder.AddDetector(new ProcessDetector());
+        return builder.AddDetector(new ProcessDetector(options));
+    }
+
+    /// <summary>
+    /// Enables process resource detector.
+    /// </summary>
+    /// <param name="builder">The <see cref="ResourceBuilder"/> being configured.</param>
+    /// <param name="configure">Optional callback action for configuring <see cref="ProcessDetectorOptions"/> which controls the behavior of the resource detector.</param>
```

Comment:
I wouldn't call this optional because it's not an optional parameter - it's just that it allows null.

Personally, I don't think it makes sense to allow null here. There's an overload that doesn't take an options, so you'd do `AddProcessDetector()` rather than `AddProcessDetector(null)` (and that's ambiguous unless you know the type of assign null to a variable of type `Action<ProcessDetectorOptions>?`).